Added error translation for error code 2571 (#41)
mattpocock/ts-error-translator: packages/engine/errors/2571.md

---
original: "Object is of type 'unknown'."
excerpt: "This variable is unknown so I don't know what type it or if {0} is in it."
---

This variable is unknown so I don't know what type it or if

```

> {0}

Reviewer: As you can see in the original, there isn't a `{0}` to reference here. This means that both the excerpt and body need changing to be clearer about what the intention of the error is.

Author: Yeah, my bad. I've already updated and removed the reference in the excerpt and body since I couldn't really get much details from the original error.

---
original: "Object is of type 'unknown'."

> excerpt: "This variable is unknown so I don't know what type it or if is in it."

Reviewer: ```suggestion
excerpt: "I don't know what type this object is."
```

---
original: "Object is of type 'unknown'."

> excerpt: "This variable is unknown so I don't know what type it or if is in it."

Reviewer 2: "unknown" === "I don't know what it is" 😅 
The excerpt is a bit redundant imho.
```suggestion
excerpt: "I cannot access properties of a variable whose type I don't know"
```

The description below could be helpful at explaining the user what to do:
````
---

You need to narrow down the type of the variable before accessing its properties or assigning it to a different typed variable. For example:

```ts
const foo: unknown = "bar";

if (typeof foo === "string") { // Now I know 'foo' is of type 'string',
  const baz: string = foo;     // I can assign it to a typed variable
  foo.toUpperCase();           // and access its properties
}
```

````

Reviewer 3: way better!

> excerpt: "This variable is unknown so I don't know what type it or if is in it."
> ---
> 
> This variable is unknown so I don't know what type it or if is in it.

Reviewer 3: ```suggestion
excerpt: "This variable is unknown so I don't know what type it or if is in it."
---

This variable is unknown so I don't know what type it or if is in it.
```

I don't think duplicating the error translation is a good thing, the excerpt should be a short description of the error but the content should be a full explanation, not the exact same thing.

Also, the sentence isn't grammarly correct.
```
what type it or if is in it.
```

> ---
> 
> This variable is unknown so I don't know what type it or if is in it.

Reviewer: ```suggestion
---
You need to narrow down the type of the variable before accessing its properties or assigning it to a different typed variable. For example:

```ts
const foo: unknown = "bar";

if (typeof foo === "string") { // Now I know 'foo' is of type 'string',
  const baz: string = foo;     // I can assign it to a typed variable
  foo.toUpperCase();           // and access its properties
}
```
```
